### ISSUE TITLE
Cannot exclude modules from package and include them in a layer

### ISSUE THREAD

**Issue by pflugs30** [serverless-heaven/serverless-webpack#1075]
# This is a Bug Report / Question

## Description
I have a Serverless project where I've isolated the dependencies to a Lambda layer in order to reduce the individual Lambda sizes and speed up deployment.  I want the Lambda layer to contain all the production NPM packages used by my Lambdas.  As such, I've put all my Lambda dependencies in the `devDependencies` section of the `package.json` file.  Then, in the Lambda layer directory, I have a `package.json` file containing all the dependencies so they can be bundled together.  My code files are structured like this:
* root
  * layers
    * appsync-layer
      * package.json
      * yarn.lock
  * serverless.yml
  * package.json
  * webpack.config.js
  * yarn.lock

When I run `sls deploy`, I want the Lambdas built without any node_modules to keep them lightweight and to depend on the Lambda layer.  However, `serverless-webpack` throws an error upon building because it can't find the modules in the `dependencies` section of my `package.json` file.  I learned I needed to specifically exclude the modules using `forceExclude` in the `custom.webpack` section of `serverless.yml`.

I originally built the project three years ago, and everything used to build just fine.  I recently upgraded all packages to their latest versions, including bringing Serverless to v3.  I now notice that my Lambda layer builds differently than it did in the past.  Now when I build, the modules I want excluded from the Lambda bundles are also excluded in the Lambda layer.  I included a configuration in an attempt to include the `node_modules` folder of the Lambda layer, but the `forceExclude` continues to remove the modules.

**Is there any way I can change the configuration to include `node_modules` in the Lambda layer and not in the Lambda bundles?**

Here's a condensed version of `serverless.yml`:
```
service: platform-appsync
useDotenv: true

frameworkVersion: ">=3.0.0"

provider:
  name: aws
  runtime: nodejs12.x
  stage: ${opt:stage, self:custom.defaultStage}
  environment:
    NODE_PATH: "./:/opt/node_modules"

layers:
  appsync:
    path: layers/appsync-layer
    compatibleRuntimes:
      - nodejs12.x
    package:
      patterns:
        - node_modules/**

functions:
  databaseSync:
    handler: src/databaseSync.handler
    name: ${self:provider.stage}-databaseSync
    layers:
      - { Ref: AppsyncLambdaLayer }

plugins:
  - serverless-webpack
  - serverless-appsync-plugin
  - serverless-appsync-simulator
  - serverless-offline

custom:
  webpack:
    excludeFiles: "**/*.spec.js"
    includeModules:
      forceExclude:
        - aws-sdk
        - mysql2
        - p-iteration
        - sequelize
        - uuid
      packagePath: "./package.json"
    webpackConfig: "webpack.config.js"
    packager: "yarn"
```

The `root/layers/appsync-layer/package.json` file:
```
{
  "name": "appsync-layer",
  "description": "Contains the packages needed for the Appsync layer",
  "dependencies": {
    "aws-sdk": "2.1074.0",
    "mysql2": "2.3.3",
    "p-iteration": "1.1.8",
    "sequelize": "6.16.1",
    "uuid": "8.3.2"
  }
}
```

And also `webpack.config.js`:
```
const path = require('path');
const slsw = require('serverless-webpack');
const nodeExternals = require('webpack-node-externals');

module.exports = {
    mode: slsw.lib.webpack.isLocal ? 'development' : 'production',
    target: 'node',
    entry: slsw.lib.entries,
    externals: [nodeExternals()],
    output: {
        path: path.resolve(__dirname, '.webpack'),
        libraryTarget: 'commonjs2',
    },
    module: {
        rules: [
            {
                test: /\.js$/,
                exclude: /\.spec\.js$/,
                use: {
                    loader: 'babel-loader',
                },
            },
        ],
    },
};
```

## Additional Data
* ***Serverless-Webpack Version***: 5.6.1
* ***Webpack version***: 5.69.0
* ***Serverless Framework Version***: 3.2.1
* ***Operating System***: Win 10 x64
* ***Stack Trace (if available)***: N/A

**Comment by pflugs30** [comment]
Could be related to [this Serverless issue](https://github.com/serverless/serverless/issues/5892)

**Comment by pflugs30** [comment]
I found that the modules were being removed by the Serverless package functionality.  The packager setting `excludeDevDependencies: true` was responsible for removing the `node_modules` folder since my layer didn't contain any NodeJS code.  I only intended the layer to include the packages.  Upon looking at [the Serverless code](https://github.com/serverless/serverless/blob/bf79934e9b43a0b2e8613ee29f9430fa22c41481/lib/plugins/package/lib/zip-service.js#L35), I also noticed that the source seems to only operate on the `include` or `exclude` properties, not the `patterns` config, like the deprecation documentation says.

When I set `excludeDevDependencies: false`, the `node_modules` in my layer were correctly included per my configuration.  So, my immediate problem is solved.  I'll likely log an issue on the Serverless project.
